File: datamodel/src/main/java/datamodel/clinical/Dosage.java
package datamodel.clinical;

import org.immutables.value.Value;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
public interface Dosage {

    @Nullable
    Double dosageMin();

    @Nullable
    Double dosageMax();

    @Nullable
    String dosageUnit();

    @Nullable
    Double frequency();

    @Nullable
    String frequencyUnit();

    @Nullable
    Double periodBetweenValue();

    @Nullable
    String periodBetweenUnit();

    @Nullable
    Boolean ifNeeded();
}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/Dosage.kt
package com.hartwig.actin.datamodel.clinical

data class Dosage(
    val dosageMin: Double? = null,
    val dosageMax: Double? = null,
    val dosageUnit: String? = null,
    val frequency: Double? = null,
    val frequencyUnit: String? = null,
    val periodBetweenValue: Double? = null,
    val periodBetweenUnit: String? = null,
    val ifNeeded: Boolean? = null
)
